main.tf

```diff
@@ -1,38 +1,50 @@
-##----------------------------------------------------------------------------- 
-## Locals declaration for determining the id of ddos protection plan.    
+##-----------------------------------------------------------------------------
+## Locals declaration for determining the id of ddos protection plan.
 ##-----------------------------------------------------------------------------
 locals {
-  ddos_pp_id = var.enable_ddos_pp && var.existing_ddos_pp != null ? var.existing_ddos_pp : var.enable_ddos_pp && var.existing_ddos_pp == null ? azurerm_network_ddos_protection_plan.example[0].id : null
+  ddos_pp_id = var.enable_ddos_pp == false && var.existing_ddos_pp != null ? var.existing_ddos_pp : var.enable_ddos_pp && var.existing_ddos_pp == null ? azurerm_network_ddos_protection_plan.example[0].id : null
 }
 
-##----------------------------------------------------------------------------- 
-## Labels module callled that will be used for naming and tags.   
+##-----------------------------------------------------------------------------
+## Labels module callled that will be used for naming and tags.
 ##-----------------------------------------------------------------------------
 module "labels" {
-
-  source  = "clouddrove/labels/azure"
-  version = "1.0.0"
-
+  source      = "clouddrove/labels/azure"
+  version     = "1.0.0"
   name        = var.name
   environment = var.environment
   managedby   = var.managedby
   label_order = var.label_order
   repository  = var.repository
 }
 
-##----------------------------------------------------------------------------- 
-## Below resource will deploy virtual network in your azure environment.    
+##-----------------------------------------------------------------------------
+## Below resource will deploy virtual network in your azure environment.
 ##-----------------------------------------------------------------------------
 resource "azurerm_virtual_network" "vnet" {
   count                   = var.enable == true ? 1 : 0
   name                    = format("%s-vnet", module.labels.id)
+  address_space           = length(var.address_spaces) == 0 ? [var.address_space] : var.address_spaces
   resource_group_name     = var.resource_group_name
+  flow_timeout_in_minutes = var.flow_timeout_in_minutes
   location                = var.location
-  address_space           = length(var.address_spaces) == 0 ? [var.address_space] : var.address_spaces
   dns_servers             = var.dns_servers
   bgp_community           = var.bgp_community
   edge_zone               = var.edge_zone
-  flow_timeout_in_minutes = var.flow_timeout_in_minutes
+
+  encryption {
+    enforcement = var.enforcement
+  }
+
+  dynamic "subnet" {
```

Comment:
why this block is added?

Reply:
Specifies if the encrypted Virtual Network allows VM that does not support encryption